### ISSUE TITLE
Conditional fields

### ISSUE THREAD

**Issue by CarloBu** [sveltia/sveltia-cms#139]
I can't find the option to make Conditional fields based of the boolean status.
The idea is to make field visibility conditional, based on other fields' values.
maybe it's here, but I can't find it.

by example:
    - {label: Is News?, name: isNews, widget: boolean, default: true, i18n: duplicate}
    - {label: News Description, name: newsDescription, widget: string, i18n: true}
    
the idea that the **News Description** field would be visible only when **isNews** boolean is true. that would give a good UX for the client to see only fields that are required for the situation. In this case, the news field would be shown if the project is news worthy,
![image](https://github.com/sveltia/sveltia-cms/assets/145860771/1da3909a-4cc7-43ba-9f07-827a7d76535a)
![image](https://github.com/sveltia/sveltia-cms/assets/145860771/74c6e80a-4de1-46d1-a92c-28fca9d49650)

**Comment by kyoshino** [comment]
I think this is relevant:

- https://github.com/decaporg/decap-cms/issues/565

Sveltia CMS has already solved this problem by supporting [variable types](https://decapcms.org/docs/variable-type-widgets/) for the Object widget, not only for the List widget. I don’t have a detailed document yet, but you can use the same `types` option to have conditional fields. Check it out!

**Comment by CarloBu** [comment]
Yes, I'm familiar with this, I'm using this feature already for other collections. You are right this is kinda of working. I just thought the boolean + condition field would be a more intuitive solution for this particular case when I need an on-off state.

```
    fields:
      - label: 'Is it News?'
        name: 'isNews'
        required: false
        widget: 'list'
        types:
          - label: 'News Description'
            name: 'carousel'
            widget: string
            fields:
              - {label: News Description, name: newsDescription, widget: string, required: false}
```
              
https://github.com/sveltia/sveltia-cms/assets/145860771/d148c689-8a3e-42c2-aaa7-3829fa9f0c36

**Comment by kyoshino** [comment]
For the variable object types, you can only add only one field. Actually in your case, you can just use `required: false` with `widget: object`.

```yaml
    fields:
      - label: 'Is it News?'
        name: 'isNews'
        required: false
        widget: object
        fields:
          - label: 'News Description'
            name: 'carousel'
            widget: string
            fields:
              - {label: News Description, name: newsDescription, widget: string, required: false}
```

The UI will look like:

<img width="733" alt="image" src="https://github.com/sveltia/sveltia-cms/assets/2929505/3240f36c-f2dc-4709-a934-7b6cd29ccec6">

<img width="735" alt="image" src="https://github.com/sveltia/sveltia-cms/assets/2929505/b9eef978-bffe-4443-8807-643fc1488398">

**Comment by CarloBu** [comment]
oh, this is much better, thanks!

**Comment by kyoshino** [comment]
And yeah, as you said, a checkbox may be more intuitive here. I can change the UI 🙂

**Comment by CarloBu** [comment]
yes, but if it's only not hard to implement for you.

I guess it could be simple ternary operator to the string widget and could look like:

    fields:
      - label: Is it News?
        name: isNews
        widget: boolean
        default: true
        required: false}
      - label: News Description
        name: newsDescription
        **condition: 'isNews'**
        widget: string}

**Comment by CarloBu** [comment]
wait, you mentioned checkboxes, are checkboxes possible for relation widget?
![image](https://github.com/sveltia/sveltia-cms/assets/145860771/924eed30-edfe-4cac-b628-179cf6e1a6a9)

I went to the sveltia and decapcms issues and found only mention custom tag widget. I made that the client can build and modify categories list through reference content collection schema, and at current I can only let to select only one item through radio buttons. If it's possible to convert to checkboxes to get the array, please give me a hint.
    fields:
      - {label: Category, name: category, widget: relation, collection: categories, value_field: defaultLocaleVersion, display_fields: [category]}
      
![image](https://github.com/sveltia/sveltia-cms/assets/145860771/90ce12cc-86e3-477b-83c3-2e8fc9000c67)

**Comment by kyoshino** [comment]
You can get checkboxes with `multiple: true`.

**Comment by CarloBu** [comment]
hehe, lol, that's unexpected:
![image](https://github.com/sveltia/sveltia-cms/assets/145860771/a34afea3-7c77-43ec-b4f2-127811155b3d)
it works like a charm!
I double checked now, and I still didn't find about "multiple: true" in the sveltiacms or decap readme. 
You need proper documentation about advanced fields.
I will collect these cool options that I will found and prepare these for the template, 100% sure very few people know about these powerful options.

**Comment by kyoshino** [comment]
I think the Decap [Relation widget](https://decapcms.org/docs/widgets/#relation) doc mentions the `multiple` option 🙂 The checkbox UI is only with Sveltia 😆 

I just shipped [v0.26.2](https://github.com/sveltia/sveltia-cms/releases/tag/v0.26.2) to use a checkbox for optional objects. I’ll work on conditional fields sometime later.

<img width="736" alt="image" src="https://github.com/sveltia/sveltia-cms/assets/2929505/9d800028-c938-4189-a9b3-6706596210a9">

**Comment by CarloBu** [comment]
I care a lot about the client's UX, and you make me happy! thanks!

this is better now than I expected!
https://github.com/sveltia/sveltia-cms/assets/145860771/60c1b512-b000-4712-925e-33f76f477e0d

**Comment by CarloBu** [comment]
Hey, @kyoshino kyoshino
I think I found the bug in the checkbox for optional objects when there is a nested list, it generates a duplicate in the second language. for a single language, it works fine.
```
collections:
  - name: Test
    label: testPage
    folder: src/content/test
    create: true
    i18n: true
    format: yml
    extension: yaml
    fields:
      - label: 'Test Field'
        name: 'testField'
        required: false
        widget: object
        i18n: duplicate
        fields:
          - label: 'Nested test field list'
            name: 'nestedTestFieldList'
            widget: list
            i18n: duplicate
            fields:
              - {label: Title, name: title, widget: string, i18n: true}
```

https://github.com/sveltia/sveltia-cms/assets/145860771/bd6bdc6c-6e51-4c77-a534-d4f6304534e5

**Comment by CarloBu** [comment]
also another strange behavior with the setup below.
if the title has i18n true:
I get the testField null or object.
the yaml output:
```
title: sample
testField: null
```
setup:
```
collections:
  - name: Test
    label: testPage
    folder: src/content/test
    create: true
    i18n: duplicate
    format: yml
    extension: yaml
    fields:
      - {label: Title, name: title, widget: string, i18n: true}   <---   i18n: true
      - label: 'Test Field'
        name: 'testField'
        required: false
        widget: object
        i18n: duplicate
        fields:
          - label: 'Nested test field list'
            name: 'nestedTestFieldList'
            widget: list
            i18n: duplicate
            fields:
              - {label: Title, name: title, widget: string, i18n: true}
```
but when the title has i18n duplicate,
I get the testField either present or object.
the yaml output:
```
title: sample
```

setup:
```
collections:
  - name: Test
    label: testPage
    folder: src/content/test
    create: true
    i18n: duplicate
    format: yml
    extension: yaml
    fields:
      - {label: Title, name: title, widget: string, i18n: duplicate} <---  i18n: duplicate
      - label: 'Test Field'
        name: 'testField'
        required: false
        widget: object
        i18n: duplicate
        fields:
          - label: 'Nested test field list'
            name: 'nestedTestFieldList'
            widget: list
            i18n: duplicate
            fields:
              - {label: Title, name: title, widget: string, i18n: true}
```

**Comment by kyoshino** [comment]
Will check.

**Comment by kyoshino** [comment]
The fix shipped with [v0.35.0](https://github.com/sveltia/sveltia-cms/releases/tag/v0.35.0) 🚢

**Comment by CarloBu** [comment]
super, thank you very much! Will check how it works asap
